scaleway/resource_iam_ssh_key.go

```diff
@@ -0,0 +1,150 @@
+package scaleway
+
+import (
+	"context"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	iam "github.com/scaleway/scaleway-sdk-go/api/iam/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func resourceScalewayIamSSKKey() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScalewayIamSSKKeyCreate,
+		ReadContext:   resourceScalewayIamSSHKeyRead,
+		UpdateContext: resourceScalewayIamSSKKeyUpdate,
+		DeleteContext: resourceScalewayIamSSKKeyDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The name of the iam SSH key",
+			},
+			"public_key": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The public SSH key",
+				// We don't consider trailing \n as diff
+				DiffSuppressFunc: func(k, oldValue, newValue string, d *schema.ResourceData) bool {
+					return strings.Trim(oldValue, "\n") == strings.Trim(newValue, "\n")
+				},
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time of the creation of the iam SSH Key",
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time of the last update of the iam SSH Key",
+			},
+			"organization_id": organizationIDSchema(),
+			"project_id":      projectIDSchema(),
+			"disabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Computed:    true,
+				Description: "The SSH key status",
+			},
+		},
+	}
+}
+
+func resourceScalewayIamSSKKeyCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	iamAPI := iamAPI(meta)
+
+	res, err := iamAPI.CreateSSHKey(&iam.CreateSSHKeyRequest{
+		Name:      d.Get("name").(string),
+		PublicKey: strings.Trim(d.Get("public_key").(string), "\n"),
+		ProjectID: (d.Get("project_id")).(string),
+	}, scw.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if _, disabledExists := d.GetOk("disabled"); disabledExists {
+		_, err = iamAPI.UpdateSSHKey(&iam.UpdateSSHKeyRequest{
+			SSHKeyID: d.Id(),
+			Disabled: expandBoolPtr(d.Get("disabled")),
+		}, scw.WithContext(ctx))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	d.SetId(res.ID)
+
+	return resourceScalewayIamSSHKeyRead(ctx, d, meta)
+}
+
+func resourceScalewayIamSSHKeyRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
Many attributes are missing from the schema. Should the `Fingerprint` be exposed?